wagtail/wagtailcore/models.py

```diff
@@ -587,32 +588,12 @@ def check(cls, **kwargs):
         return errors
 
     def _update_descendant_url_paths(self, old_url_path, new_url_path):
-        cursor = connection.cursor()
-        if connection.vendor == 'sqlite':
-            update_statement = """
-                UPDATE wagtailcore_page
-                SET url_path = %s || substr(url_path, %s)
-                WHERE path LIKE %s AND id <> %s
-            """
-        elif connection.vendor == 'mysql':
-            update_statement = """
-                UPDATE wagtailcore_page
-                SET url_path= CONCAT(%s, substring(url_path, %s))
-                WHERE path LIKE %s AND id <> %s
-            """
-        elif connection.vendor in ('mssql', 'microsoft'):
-            update_statement = """
-                UPDATE wagtailcore_page
-                SET url_path= CONCAT(%s, (SUBSTRING(url_path, 0, %s)))
-                WHERE path LIKE %s AND id <> %s
-            """
-        else:
-            update_statement = """
-                UPDATE wagtailcore_page
-                SET url_path = %s || substring(url_path from %s)
-                WHERE path LIKE %s AND id <> %s
-            """
-        cursor.execute(update_statement, [new_url_path, len(old_url_path) + 1, self.path + '%', self.id])
+        (Page.objects
+            .filter(path__startswith=self.path)
+            .exclude(pk=self.pk)
+            .update(url_path=Concat(
+                Value(new_url_path),
+                Substr('url_path', len(old_url_path) + 1))))
```

Comment:
@atombrella has helpfully provided an independent implementation of this fix at #3895, and the one effective difference from this is that it has `F('url_path')` instead of `'url_path`. I'm pretty sure `F('url_path')` is correct (otherwise it'd be taking a substring of the literal string `'url_path'`, surely?) but we'll have to double-check this before merging either version...

Reply:
OK, the `F` isn't necessary - as per [the Django docs](https://docs.djangoproject.com/en/1.11/ref/models/expressions/#value-expressions):

> You will need to use Value() when you want to pass a string to an expression. Most expressions interpret a string argument as the name of a field, like Lower('name').

So it would have to be `Value('url_path')` to be interpreted as the literal string `'url_path'` - although that wouldn't work anyhow, because apparently Django db functions don't let you construct expressions that don't reference any database fields (it fails with `Cannot resolve expression type, unknown output_field`)...